web/src/components/pages/datasets/dataset-info.tsx

```diff
@@ -1,56 +1,61 @@
 import * as React from 'react';
 import { useEffect, useState } from 'react';
-import { CloudIcon } from '../../ui/icons';
 import { Spinner } from '../../ui/ui';
 
-import { getRelease, CURRENT_RELEASE_ID } from './releases';
-
 import DatasetIntro from './dataset-intro';
 import DatasetCorpusDownload from './dataset-corpus-download';
 import DatasetSegmentDownload from './dataset-segment-download';
-import DatasetDescription from './dataset-description';
+import { useAPI } from '../../../hooks/store-hooks';
 
 import './dataset-info.css';
+import { useLocale } from '../../locale-helpers';
+import DatasetDescription from './dataset-description';
+import { Dataset } from 'common';
 
 const DatasetInfo = () => {
   const [isLoading, setIsLoading] = useState(true);
 
-  const [releaseId, setReleaseId] = useState(CURRENT_RELEASE_ID);
-  const [releaseData, setReleaseData] = useState(null);
+  const [languagesWithDatasets, setLanguagesWithDatasets] = useState([]);
+  const [currentDataset, setCurrentDataset] = useState<Dataset>();
+
+  const api = useAPI();
+  const [globalLocale] = useLocale();
 
   useEffect(() => {
     setIsLoading(true);
-    getRelease(releaseId)
-      .then(setReleaseData)
-      .finally(() => {
-        setIsLoading(false);
-      });
-  }, [releaseId]);
+
+    //get all languages w/ dataset releases
+    api.getLanguagesWithDatasets().then(data => {
+      setLanguagesWithDatasets(data);
+    });
+
+    //get stats for latest full release
+    api.getDatasets('complete').then(data => {
+      setCurrentDataset(data[0]);
+      setIsLoading(false);
+    });
+  }, []);
```

Comment:
I think we should have a generic hook for fetching data so we won't need to replicate this logic for data fetching across components.